spec/data-model/README.md

```diff
@@ -74,28 +120,35 @@ interface CatchallKey {
 ## Patterns
 
 Each `Pattern` represents a linear sequence, without selectors.
-Each element of the sequence MUST have either a `Text` or an `Expression` shape.
-`Text` represents literal _text_,
+Each element of the `body` array MUST either be a string or an `Expression` object.
+String values represent literal _text_,
 while `Expression` wraps each of the potential _expression_ shapes.
-The `value` of `Text` is the "cooked" value (i.e. escape sequences are processed).
+The `body` strings are the "cooked" _text_ values, i.e. escape sequences are processed.
 
-Implementations MUST NOT rely on the set of `Expression` `body` values being exhaustive,
+Implementations MUST NOT rely on the set of `Expression` interfaces being exhaustive,
 as future versions of this specification MAY define additional expressions.
-A `body` with an unrecognized value SHOULD be treated as an `Unsupported` value.
+An `Expression` `func` with an unrecognized value SHOULD be treated as an `UnsupportedExpression` value.
 
 ```ts
 interface Pattern {
-  body: Array<Text | Expression>;
+  body: Array<string | Expression>;
 }
 
-interface Text {
-  type: "text";
-  value: string;
+type Expression = LiteralExpression | VariableExpression | FunctionExpression;
+
+interface LiteralExpression {
+  arg: Literal;
+  func?: FunctionRef | UnsupportedExpression;
+}
```

Comment:
For consistency with the data model implied by the grammar, `func` fields should probably be renamed (and possibly `FunctionExpression` as well, in both the grammar and here).
```suggestion
type Expression = LiteralExpression | VariableExpression | AnnotationExpression;

interface LiteralExpression {
  arg: Literal;
  annotation?: FunctionRef | UnsupportedAnnotation;
}
```
etc.

Reply:
One challenge with "annotation" is that when it's in an expression without an operand, what is it "annotating"?

Reply:
The expression.

Reply:
For the cases where the implementation does support a specific private-use annotation, we have this: https://github.com/unicode-org/message-format-wg/blob/f35985614c66eb513954f342fcebc1532530ae02/spec/data-model/README.md?plain=1#L167-L171

And then in the [Extensions](https://github.com/unicode-org/message-format-wg/blob/main/spec/data-model/README.md#extensions) section, we specify that an implementation is allowed to extend the data model for that purpose. Or at least it's intended to do so.

So we _can_ at this point lump all of the unsupported reserved & private-use into this one basket of the data model, rather than needing to split it as in the syntax.

Reply:
If the data-model is meant to be interchanged, does that mean that the "private-use-ness" might be a relevant detail to the consumer? I mean, maybe it's `FunctionRef` but if it weren't, it wouldn't be `UnsupportedAnnotation` necessarily either? Or am I barking up the wrong tree? 🌳

Reply:
If an implementation does define and support a private-use annotation, I think there are two options for its data model representation:
1. It's mapped to some representation within the base data model, e.g. as some `func` value or maybe as an operand of some sort. But then why need the new syntax in the first place?
2. The annotation is doing something truly different, and it doesn't fit within the base data model. In this case, the implementation extends the base, e.g. adding a new field in the `Expression` interfaces.

If going the first route, then interchange is of course easier, but that syntax will also have some representation without private use. If going the second route, interchange will require both ends to agree about the private-use.